fitapp/tasks.py

```diff
@@ -87,22 +86,25 @@ def get_time_series_data(fitbit_user, cat, resource, date=None):
                     tsd.save()
             # Release the lock
             cache.delete(lock_id)
-    except HTTPTooManyRequests:
+    except HTTPTooManyRequests as e:
         # We have hit the rate limit for the user, retry when it's reset,
         # according to the reply from the failing API call
-        e = sys.exc_info()[1]
         logger.debug('Rate limit reached, will try again in %s seconds' %
                      e.retry_after_secs)
-        raise get_time_series_data.retry(exc=e, countdown=e.retry_after_secs)
-    except HTTPBadRequest:
+        raise get_time_series_data.retry(
+            exc=e,
+            countdown=e.retry_after_secs + int(
+                # Exponential back-off + random jitter
+                random.uniform(2, 4) ** self.request.retries
+            )
+        )
+    except HTTPBadRequest as exc:
```

Comment:
@brad Sorry for the nitpick, can we be more consistent with exception naming here? (Using `e` everywhere or `exc`)